### PR TITLE
Fixes for intersections and coverings

### PR DESCRIPTION
**Pull request by simonbrandhorst**
Its not perfect, but it is an improvement.

### CODE REVIEW

File: experimental/Schemes/AffineSchemes.jl [hunk 1]
```diff
@@ -824,6 +826,9 @@ function restrict(f::SpecMor, U::AbsSpec, V::AbsSpec; check::Bool=true)
   return SpecMor(U, V, OO(U).(pullback(f).(gens(domain(pullback(f))))), check=check)
 end
 
+restriction(f::SpecMor, U::AbsSpec, V::AbsSpec; check::Bool=true) = restrict(f,U,V,check=check)
```
Comment (thofma) [inline experimental/Schemes/AffineSchemes.jl:829]: Why the alias?

Comment (simonbrandhorst) [reply experimental/Schemes/AffineSchemes.jl:829]: Some other parts of the code expected a `restriction` though `restrict`  seems to be the more widely used term to obtain the restriction of something.  I did not want to mess around too much.

Comment (thofma) [reply experimental/Schemes/AffineSchemes.jl:829]: OK, understood. This should be cleaned up at some point. Maybe we can leave a TODO.

File: experimental/Schemes/CoveredScheme.jl [hunk 2]
```diff
@@ -285,29 +285,25 @@ end
...
+      Ui = PrincipalOpenSubset(U[i], OO(U[i])(x[j-1]))
+      Uj = PrincipalOpenSubset(U[j], OO(U[j])(y[i]))
+      f = SpecMor(Ui, Uj,
                       vcat([x[k]//x[j-1] for k in 1:i-1],
                            [1//x[j-1]],
                            [x[k-1]//x[j-1] for k in i+1:j-1],
                            [x[k]//x[j-1] for k in j:r],
                            x[r+1:end]),
-                      check=false
+                      check=true
```
Comment (HechtiDerLachs) [inline experimental/Schemes/CoveredScheme.jl]: This should be set back to false.

File: experimental/Schemes/CoveredScheme.jl [hunk 3]
```diff
@@ -146,22 +146,22 @@ function intersect_in_covering(U::AbsSpec, V::AbsSpec, C::Covering)
...
+      f = gens(affine_refinements(C)[X][j][1])[k]
     end
-    g = one(base_ring(OO(X)))
+    g = one(ambient_ring(X))
     if m != 0
-      g = gens(affine_refinements(C)[C[l]][m][1])[n]
+      g = gens(affine_refinements(C)[X][m][1])[n]
     end
-    W = SpecOpen(X, [f*g])
+    W = PrincipalOpenSubset(X, [f*g])
```
Comment (HechtiDerLachs) [inline experimental/Schemes/CoveredScheme.jl:161]: Let's try to find a test for this line, tomorrow. Also, we should run the whole method on covered schemes that use both types of glueings.

Comment (simonbrandhorst) [reply experimental/Schemes/CoveredScheme.jl:161]: We can write it when #1573 is merged.

**Review by afkafkafk13 — APPROVED** [review]
small pull request, looks good
- fixes make sense
- parts, which are in transitional status, are marked by appropriate comments

**Pull request merged**